### PR TITLE
feat: filter functionality and auth 

### PR DESCRIPTION
**Pull request by sri-rishi**
### files to check

- auth.context.js
- Authentication 
- postMethodRequest.js
- UserProfile.jsx

### CODE REVIEW

File: src/ApiCalls/getMethodRequest.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import axios from "axios";
+
+const getData = async(setVideosList) => {
```
Comment (bhtibrewal) [inline src/ApiCalls/getMethodRequest.js:3]: Suggestion: this should be getVideoList

File: src/Components/Navbar/Navbar.jsx [hunk 2]
```diff
@@ -1,29 +1,34 @@
+import { NavLink } from "react-router-dom";
 import { GiHamburgerMenu } from "../../assets";
+import { useAuth } from "../../Context";
 import { Button } from "../index";
 import { SearchBar } from "./SearchBar";
 
 export const Navbar = () => {
+    const {isUserLoggedIn, user} = useAuth();
```
Comment (bhtibrewal) [inline src/Components/Navbar/Navbar.jsx:8]: appreciation: `isUserLoggedIn` liked the variable naming here
suggestion: do something similar for `user` also

File: src/Helper/filterFunciton.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const getFilteredBySearchData = (videoList, searchValue) => {
+    if(searchValue) {
+        return videoList.filter(video => video.title.toLowerCase().includes(searchValue))
```
Comment (bhtibrewal) [inline src/Helper/filterFunciton.js:3]: Suggestion: do lowercase to searchValue also

File: src/Components/Navbar/SearchBar.jsx [hunk 4]
```diff
@@ -1,14 +1,30 @@
...
+                    onChange={(e) => setSearchValue(e.target.value)}
+                />
             </label>
-            <Button className="nav-search-btn btn-border-none bg-transparent" type="submit" icon={<FaSearch className="icon-vr-align"/>}/>       
+            <Button 
+                className="nav-search-btn btn-border-none bg-transparent" 
+                type="submit" 
+                icon={<FaSearch className="icon-vr-align"/>}
+                onClick={() => filterDispatch({type: "FILTER_BY_SERACH", payload: searchValue})}
+            />
```
Comment (nikhiljugale007) [inline src/Components/Navbar/SearchBar.jsx:27]: suggestion: Try to create a separate function for handling onclick events.

**Review by nikhiljugale007 — APPROVED** [review]
Did not find any major issue. Good to merge.
UI suggestion: You can highlight chip which is selected in filtering.

**Review by bhtibrewal — APPROVED** [review]
just few small suggestions else good to merge

**Review by Sahil25061999 — COMMENTED** [review]
code looks good, loved the file structure,
just a problem while sign up, if i don't give any input and check on "terms and condition" checkbox, I am able to sign in.

**Pull request merged**